dsc/docs-conceptual/dsc-2.0/tutorials/create-dsc-resource-machine-config.md

```diff
@@ -0,0 +1,2086 @@
+
+It also specifies that `$ConfigurationScope`'s type is **TailspinScope**. To define the
+**TailspinScope** type, add the following **TailspinScope** enum after the class definition in
+`ExampleResources.psm1`:
+
+```powershell
+enum TailspinScope {
+    Machine
+    User
+}
+```
+
+This enumeration makes `Machine` and `User` the only valid values for the `$ConfigurationScope`
+property of the DSC Resource.
+
+### Add the Ensure property
+
+It's best practice to define an `$Ensure` property to control whether an instance of a DSC Resource
+exists. An `$Ensure` property usually has two valid values, `Absent` and `Present`.
+
+- If `$Ensure` is specified as `Present`, the DSC Resource creates the item if it doesn't exist.
+- If `$Ensure` is `Absent`, the DSC Resource deletes the item if it exists.
+
+For the **Tailspin** DSC Resource, the item to create or delete is the configuration file for the
+specified `$ConfigurationScope`.
+
+Define **TailspinEnsure** as an enum after **TailspinScope**. It should have the values `Absent` and
+`Present`.
+
+```powershell
+enum TailspinEnsure {
+    Absent
+    Present
+}
+```
+
+Next, add the `$Ensure` property in the class after the `$ConfigurationScope` property. It should
+have an empty **DscProperty** attribute and its type should be **TailspinEnsure**. It should default
+to `Present`.
+
+```powershell
+[DscProperty()] [TailspinEnsure]
+$Ensure = [TailspinEnsure]::Present
+```
+
+### Add the UpdateAutomatically property
+
+To manage automatic updates, define the `$UpdateAutomatically` property in the class after the
+`$Ensure` property. Its **DscProperty** attribute should indicate that it's mandatory and its type
+should be **boolean**.
+
+```powershell
+[DscProperty(Mandatory)] [bool]
+$UpdateAutomatically
+```
+
+### Add the UpdateFrequency property
+
+To manage how often TSToy should check for updates, add the `$UpdateFrequency` property in the
+class after the `$UpdateAutomatically` property. It should have an empty **DscProperty** attribute
+and its type should be **int**. Use the **ValidateRange** attribute to limit the valid values for
+`$UpdateFrequency` to between 1 and 90.
+
+```powershell
+[DscProperty()] [int] [ValidateRange(1, 90)]
+$UpdateFrequency
+```
+
+### Add the Reasons property
+
+Because this DSC Resource is intended for use with Azure Automanage's machine configuration
+feature, it must have **Reasons** property that meets the following requirements:
+
+- It must be declared with the **NotConfigurable** property on the **DscProperty** attribute.
+- It must be an array of objects that have a **String** property named **Code**, a **String**
+  property named **Phrase**, and no other properties.
+
+Machine configuration uses the **Reasons** property to standardize how compliance information is
+presented. Each object returned by the `Get()` method for the **Reasons** property identifies how
+and why an instance of the DSC Resource isn't compliant.
+
```

Comment:
```suggestion

Resources should always return a Reasons object, even when the result is Compliant. End users
might not understand the script or how it works, so the results should prove the check was run,
what was checked, and what was found.

As an example phrase:

```text
Expected the UpdateFrequency property in tstoy.config.json to have a value of 45, but the value was 30.
```

```

Reply:
Should they always return a **Reasons** object or should they always have a **Reasons** property?

I'm not sure how to interpret this statement, given **Reasons** only populates (afaict) when something is non-compliant:

> End users might not understand the script or how it works, so the results should prove the check was run, what was checked, and what was found.